### PR TITLE
add module

### PR DESCRIPTION
**Pull request by siervo-jallaine**
Create a class with atleast 5 methods (excluding constructor) and 3 properties, adding a menu method.

### CODE REVIEW

File: herchel/siervo.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    def __init__(self):
+        self.students = {}  
+        self.subjects = [] 
+        self.max_grade = 100 
+
+    def add_grade(self):
+        name = input("Enter the student's name: ").strip().lower()
+        if name not in self.students:
+            self.students[name] = {}
+        subject = input("Enter the subject: ").strip().lower()
```
Comment (Riomalos-ZyrrahFeil) [inline herchel/siervo.py:12]: add space before subject variable (line 11).

File: herchel/siervo.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                print(f"Invalid. Must be between 0 and {self.max_grade}.")
+        except ValueError:
+            print("Invalid input. Grade must be a number.")
+
+    def view_grades(self):
+        name = input("Enter the student's name: ").strip().lower()
+        if name not in self.students:
+            print("Student not found.")
+            return
+        print(f"Grades for {name}:")
```
Comment (Riomalos-ZyrrahFeil) [inline herchel/siervo.py:32]: add space before and after the print (line 30).

File: herchel/siervo.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        else:
+            for subject, grade in grades.items():
+                print(f"  {subject}: {grade}")
+ 
+    def update_grade(self):
+        name = input("Enter the student's name: ").strip().lower()
+        if name not in self.students:
+            print("Student not found.")
+            return
+        subject = input("Enter subject to update: ").strip().lower()
```
Comment (Riomalos-ZyrrahFeil) [inline herchel/siervo.py:47]: add space before subject variable (line 44).

File: herchel/siervo.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                print(f"Invalid. Must be between 0 and {self.max_grade}.")
+        except ValueError:
+            print("Invalid input. Grade must be a number.")
+
+    def calculate_average(self):
+        name = input("Enter the student's name: ").strip().lower()
+        if name not in self.students:
+            print("Student not found.")
+            return
+        grades = [grade for grade in self.students[name].values()
```
Comment (Riomalos-ZyrrahFeil) [inline herchel/siervo.py:68]: add space before grades variable (line 64).

File: herchel/siervo.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        else:
+            average = sum(grades) / len(grades)
+            print(f"Average grade for {name}: {average:.2f}")
+
+    def delete_grades(self):
+        name = input("Enter Student's Name: ").strip().lower()
+        if name not in self.students:
+            print("Student not found.")
+            return
+        subject = input("Enter the subject to delete grades for: "
```
Comment (Riomalos-ZyrrahFeil) [inline herchel/siervo.py:83]: add space before subject variable (line 78).

File: herchel/siervo.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            print("Invalid input. Grade must be a number.")
+
+    def view_grades(self):
+        name = input("Enter the student's name: ").strip().lower()
+        if name not in self.students:
+            print("Student not found.")
+            return
+        print(f"Grades for {name}:")
+        grades = self.students[name]
+
```
Comment (Riomalos-ZyrrahFeil) [inline herchel/siervo.py]: remove line 32 space.

**Review by Riomalos-ZyrrahFeil — CHANGES_REQUESTED** [review]
I will approve this PR once the changes are completed

File: herchel/siervo.py (hunk 1 above)
Comment (siervo-jallaine) [reply herchel/siervo.py:12]: add necessary space, fix in commit 6bfb109 .

File: herchel/siervo.py (hunk 2 above)
Comment (siervo-jallaine) [reply herchel/siervo.py:32]: add necessary space, fix in commit 6bfb109 .

File: herchel/siervo.py (hunk 3 above)
Comment (siervo-jallaine) [reply herchel/siervo.py:47]: add necessary space, fix in commit 6bfb109 .

File: herchel/siervo.py (hunk 4 above)
Comment (siervo-jallaine) [reply herchel/siervo.py:68]: add necessary space, fix in commit 6bfb109 .

File: herchel/siervo.py (hunk 5 above)
Comment (siervo-jallaine) [reply herchel/siervo.py:83]: add necessary space, fix in commit 6bfb109 .

File: herchel/siervo.py (hunk 6 above)
Comment (siervo-jallaine) [reply herchel/siervo.py]: remove unnecessary space, fix in commit 6bfb109 .

**Review by Riomalos-ZyrrahFeil — APPROVED** [review]
Looks good.

File: herchel/siervo.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
+class GradeManagementSystem:
+    def __init__(self):
+        self.students = {}  
+        self.subjects = [] 
+        self.max_grade = 100
```
Comment (Riomalos-ZyrrahFeil) [inline herchel/siervo.py]: capitalize constant name - max_grade

**Review by Riomalos-ZyrrahFeil — CHANGES_REQUESTED** [review]
- capitalize constant name

File: herchel/siervo.py (hunk 7 above)
Comment (siervo-jallaine) [reply herchel/siervo.py]: change into capital letters for constant, fix in commit ac2f4b9 .

**Review by Riomalos-ZyrrahFeil — APPROVED** [review]
Much better.

**Pull request merged**